Topics/Development_Process/Serverless_AWS_Lambda.md

```diff
@@ -0,0 +1,69 @@
+# Serverless Computing with AWS Lambda
+
+## Table of Contents
+
+### [Introduction](#introduction-1)
+
+### [Benefits of Serverless Computing](#benefits-of-serverless-computing-1)
+
+### [Disadvantages of Serverless Computing](#disadvantages-of-serverless-computing-1)
+
+### [AWS Lambda](#aws-lambda-1)
+
+#### [Creating Lambda Functions in the AWS Console](#creating-lambda-functions-in-the-aws-console-1)
+
+#### [Invoking Lambda Functions from the AWS Console](#invoking-lambda-functions-from-the-aws-console-1)
+
+### [Additional Resources](#additional-resources-1)
+
+## Introduction
+
+It can be hard to set up and maintain your backend services. There are a ton of hosting options to choose from, and they can get pretty pricey.
+
+Serverless computing allows you to to build and run applications without needing to manage servers. Cloud providers such as AWS, Azure, Google Cloud automatically provision servers based on demand.
```

Comment:
You could potentially elaborate a little bit more on what exactly serverless computing is. I understand that it's when you deploy something without hosting a server yourself, but what does that actually mean? Is this some kind of interface, do you SSH into it, etc? (These are some questions I had when I read this, I think you could potentially explain serverless computing more in depth for someone who is a newcomer to it)

Reply:
Do you mean a more high-level introduction to the real idea behind serverless computing, and not its applications/use cases?

Reply:
Yeah, potentially talk a bit more about how serverless works when you use it. Is it done through some sort of no-code interface, or is it a code-heavy tool?

Reply:
Serverless is just the cloud provider running your code on their server whenever you make a request to it, without you having to worry about server resources at all. They will automatically scale up resources when there's a lot of requests, and scale down to 0 when its not being used. Would this make sense if I added it to the introduction part of the page?

Reply:
Yes that would be great to add! It clarifies how it works, especially if you put this around where you talk about how to code runs due to event triggers (because my follow up question would be how does it know when to run your code).

Reply:
Lambda itself is basically a function that can be called by an event source (console, CLI, SDK, other services), and the code is run on an AWS server somewhere. When the Lambda function is triggered, it has to be started up (which I go over in one of the disadvantage points). I will add a point about exactly what a lambda function is!